src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/KVStoRCMigrationActorStateProvider.cs

```diff
@@ -382,6 +385,172 @@ public async Task<long> SaveStateAsync(List<KeyValuePair> kvsData, CancellationT
             return keysMigrated.Count;
         }
 
+        /// <summary>
+        /// Gets Value for given Key from RC state provider
+        /// </summary>
+        /// <param name="key">Key to fetch</param>
+        /// <param name="cancellationToken">cancellationToken</param>
+        /// <returns>Value for given key</returns>
+        public async Task<byte[]> GetValueByKeyAsync(string key, CancellationToken cancellationToken)
+        {
+            using (var tx = this.rcStateProvider.GetStateManager().CreateTransaction())
+            {
+                IReliableDictionary2<string, byte[]> dictionary = null;
+                if (key.StartsWith(ActorPresenceKeyPrefix))
+                {
+                    dictionary = this.rcStateProvider.GetActorPresenceDictionary();
+                }
+
+                // It is not right to assume the ActorId wouldn't have underscores.
+                // TODO: Handle this in ambiguous ActorId PR
+                else if (key.StartsWith(ActorStorageKeyPrefix))
+                {
+                    var startIndex = this.GetNthIndex(key, '_', 2);
```

Comment:
NIT: Since figuring the dictionary is needed by more than one method, you can move this to a separate method for reusability.